File: USACO Training/3.3/Home on the Range/range.py
"""
ID: cormac.1
LANG: PYTHON3
TASK: range
"""

with open("range.in") as file:
    N = int(file.readline())
    land = [[1 if char == '1' else 0 for char in file.readline()]
            for j in range(N)]

landSum = [[0 for i in range(N+1)] for j in range(N+1)]

for y in range(1, N+1):
    for x in range(1, N+1):
        landSum[y][x] = land[y-1][x-1] + landSum[y-1][x] + landSum[y][x-1] - landSum[y-1][x-1]

results = []
for landSize in range(2, N+1):
    freq = 0
    for sx in range(1, N+2 - landSize):
        for sy in range(1, N+2 - landSize):
            area = landSum[sy+landSize-1][sx+landSize-1] - landSum[sy-1][sx+landSize-1] - landSum[sy+landSize-1][sx-1]+landSum[sy-1][sx-1]
            if area == landSize * landSize:
                freq += 1
    if freq != 0: results.append((landSize, freq))
    
with open("range.out", "w") as file:
    for size, freq in results:
        file.write(str(size) + " " + str(freq) + "\n")

File: USACO Training/3.3/Home on the Range - JAVA/src/range.java
/*
ID: cormac.1
LANG: JAVA
TASK: range
 */

import java.io.*;
import java.lang.*;
import java.util.*;

public class range {
    public static void main(String[] args) throws IOException {
        BufferedReader fin = new BufferedReader(new FileReader("range.in"));
        PrintWriter fout = new PrintWriter(new BufferedWriter(new FileWriter("range.out")));

        StringTokenizer s = new StringTokenizer(fin.readLine());

        int N = Integer.parseInt(s.nextToken());
        int land[][] = new int[N][N];

        for (int i = 0; i < N; ++i) {
            int j = 0;
            for (char c : fin.readLine().toCharArray()) {
                land[i][j] = (c == '1') ? 1 : 0;
                ++j;
            }
        }

        int landSum[][] = new int[N+1][N+1];

        for (int y = 1; y <= N; ++y) {
            for (int x = 1; x <= N; ++x) {
                landSum[y][x] = land[y-1][x-1] + landSum[y-1][x] + landSum[y][x-1] - landSum[y-1][x-1];
            }
        }

        for (int squareSize = 2; squareSize <= N; ++squareSize) {
            int freq = 0;
            for (int sy = 1; sy + squareSize - 1 <= N; ++sy) {
                for (int sx = 1; sx + squareSize - 1 <= N; ++sx) {
                    int area = landSum[sy + squareSize - 1][sx + squareSize - 1] -
                            landSum[sy-1][sx + squareSize - 1] -
                            landSum[sy + squareSize - 1][sx-1] +
                            landSum[sy-1][sx-1];
                    if (area == squareSize * squareSize) ++freq;
                }
            }
            if (freq != 0) {
                fout.println(squareSize + " " + freq);
            }
        }

        fout.close();
    }
}
